packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+    public
+  {
+    lockedGold.unlock(pendingWithdrawalValue);
+    vm.expectRevert("Requested value larger than pending value");
+    lockedGold.relock(index, pendingWithdrawalValue + 1);
+  }
+
+  function test_ShouldUpdateDelegatorDelegateeAmount_WhenDelegating() public {
+    vm.prank(delegatee);
+    accounts.createAccount();
+    lockedGold.delegateGovernanceVotes(delegatee, FixidityLib.newFixedFraction(1, 1).unwrap());
+    lockedGold.unlock(pendingWithdrawalValue / 2);
+
+    (uint256 expected, uint256 real) = lockedGold.getDelegatorDelegateeExpectedAndRealAmount(
+      address(this),
+      delegatee
+    );
+
+    assertEq(expected, pendingWithdrawalValue / 2);
+    assertEq(real, pendingWithdrawalValue / 2);
+
+    lockedGold.relock(index, pendingWithdrawalValue / 2);
+
+    (uint256 expected2, uint256 real2) = lockedGold.getDelegatorDelegateeExpectedAndRealAmount(
+      address(this),
+      delegatee
+    );
+
+    assertEq(expected2, pendingWithdrawalValue);
+    assertEq(real2, pendingWithdrawalValue);
+  }
+
+  function test_ShouldRevert_WhenPendingWithdrawalDoesNotExists() public {
+    vm.expectRevert("Bad pending withdrawal index");
+    lockedGold.relock(0, pendingWithdrawalValue);
+  }
+}
+
+contract Withdraw is LockedGoldFoundryTest {
+  uint256 value = 1000;
+  uint256 index = 0;
+
+  function setUp() public {
+    super.setUp();
+    lockedGold.lock.value(value)();
+  }
+
+  function test_ShouldRemoveThePendingWithdrawal_WhenItIsAFterTheAvailabilityTime() public {
+    lockedGold.unlock(value);
+    vm.warp(block.timestamp + unlockingPeriod + 1);
+    lockedGold.withdraw(index);
+    (uint256[] memory vals, uint256[] memory timestamps) = lockedGold.getPendingWithdrawals(
+      address(this)
+    );
+    assertEq(vals.length, 0);
+    assertEq(timestamps.length, 0);
+  }
+
+  function test_ShouldEmitGoldWithdrawnEvent_WhenItIsAfterTheAvailabilityTime() public {
+    lockedGold.unlock(value);
+    vm.warp(block.timestamp + unlockingPeriod + 1);
+    vm.expectEmit(true, true, true, true);
+    emit GoldWithdrawn(address(this), value);
+    lockedGold.withdraw(index);
+  }
+
+  function test_ShouldRevert_WhenItIsBeforeTheAvailabilityTime() public {
+    lockedGold.unlock(value);
+    vm.expectRevert("Pending withdrawal not available");
+    lockedGold.withdraw(index);
+  }
+
+  function test_ShouldRevert_WhenPendingWithdrawalDoesNotExist() public {
+    vm.expectRevert("Bad pending withdrawal index");
+    lockedGold.withdraw(index);
+  }
+
+  function() external payable {}
+}
+
+contract AddSlasher is LockedGoldFoundryTest {
+  address downtimeSlasher = actor("DowntimeSlasher");
+
+  function setUp() public {
+    super.setUp();
+    registry.setAddressFor("DowntimeSlasher", downtimeSlasher);
+  }
+
+  function test_ShouldBeAbleToAddSlasherToWhitelist() public {
+    string memory slasherName = "DowntimeSlasher";
```

Comment:
would move `slasherName = "DowntimeSlasher";`to the contract storage and use the variable in all the tests